lib/omnibus/packagers/rpm.rb

```diff
@@ -594,5 +594,29 @@ def safe_architecture
         Ohai['kernel']['machine']
       end
     end
+
+    #
+    # Install the specified packages
+    #
+    # @return [void]
+    #
+    def install(packages, enablerepo = NULL)
+      if null?(enablerepo)
+        enablerepo_string = ''
+      else
+        enablerepo_string = "--disablerepo='*' --enablerepo='#{enablerepo}'"
+      end
+      `yum check-update`
```

Comment:
I'm pretty sure `check-update` doesn't actually direct yum to clean its local cache (unlike `apt`, `yum` has an internal cache expiration mechanism and while I'm not exactly sure of how it works, I've done some local testing and `check-update` doesn't seem to re-download the package lists all the time).
What we could do instead would be to expire yum's cache with `yum clean expire-cache`, and then `yum` will update its cache when the `install` command is run (and if `disablerepo`/`enablerepo` are provided with the `install` command `yum` will only have to update these repos)